src/test/kotlin/com/contentful/java/cma/CMAHttpExceptionTest.kt

```diff
@@ -0,0 +1,232 @@
+    fun testWrongType() {
+        val request = Request.Builder().url("https://example.com/foo").build()
+
+        val response = buildMockedResponse(request, TestUtils.fileToString("error_wrong_type.json"))
+
+        val e = CMAHttpException(request, response).errorBody
+
+        assertEquals("InvalidEntry", e.sys.id)
+        assertEquals("8cd7ef1751dd283ee230e6a58efef8ec", e.requestId)
+
+        assertEquals("Validation error", e.message)
+
+        assertEquals(1, e.details.errors.size)
+
+        assertEquals("type", e.details.errors[0].name)
+        val path = e.details.errors[0].path as List<*>
+        assertEquals("fields", path[0])
+        assertEquals("xqZFfK7YdzEfhhJg", path[1])
+        assertEquals("en-US", path[2])
+        assertEquals("bar", e.details.errors[0].value)
+        assertEquals("Object", e.details.errors[0].type)
+        assertEquals("The type of \"value\" is incorrect, expected type: Object", e.details.errors[0].details)
+    }
+
+    @test
+    fun testWrongField() {
+        val request = Request.Builder().url("https://example.com/foo").build()
+
+        val response = buildMockedResponse(request, TestUtils.fileToString("error_wrong_field.json"))
+
+        val e = CMAHttpException(request, response).errorBody
+
+        assertEquals("InvalidEntry", e.sys.id)
+        assertEquals("fieldsRequestId", e.requestId)
+
+        assertEquals("The body you sent contains an unknown key.", e.message)
+
+        assertEquals(1, e.details.keys.size)
+        assertEquals("a_field_that_does_not_exist", e.details.keys[0])
+    }
+
+    @test
+    fun testToString() {
+        val request = Request.Builder().url("https://example.com/foo").build()
+
+        val response = buildMockedResponse(request, TestUtils.fileToString("error_wrong_type.json"))
+
+        val e = CMAHttpException(request, response)
+
+        assertEquals("FAILED ErrorBody { details = Details { errors = [Error " +
+                "{ details = The type of \"value\" is incorrect, expected type: Object," +
+                " name = type, path = [fields, xqZFfK7YdzEfhhJg, en-US], type = Object, " +
+                "value = bar }], }, message = Validation error, " +
+                "requestId = 8cd7ef1751dd283ee230e6a58efef8ec, sys = Sys { id = InvalidEntry, " +
+                "type = Error } }\n" +
+                "\tRequest{method=GET, url=https://example.com/foo, tag=null}\n" +
+                "\t↳ Header{}\n" +
+                "\tResponse{protocol=http/1.1, code=500, message=message, " +
+                "url=https://example.com/foo}\n" +
+                "\t↳ Header{X-Contentful-RateLimit-Hour-Limit: 1, " +
+                "X-Contentful-RateLimit-Hour-Remaining: 2, X-Contentful-RateLimit-Reset: 5k, " +
+                "X-Contentful-RateLimit-Second-Limit: 3, " +
+                "X-Contentful-RateLimit-Second-Remaining: 4}", e.toString())
+    }
+
+    private fun buildMockedResponse(request: Request?, body: String): Response? {
```

Comment:
yeah, maybe you can refactor this method to inject all the values so that when setting up each test, the author can see exactly why each expectation they are testing for is what it is.

Something like the following:

```kotlin
private fun buildMockedResponse(request: Request?, body: String, hourLimit: Int, hourRemaining: Int, secondsRemaining: Int, limitReset: Int): Response
```

Reply:
:thinking: